### PR TITLE
Chore: parse parseDuration values as numbers

### LINKED ISSUE #41558: Chore: Refactor parseDuration to return the correct type
Spotted in @thisisobate's Refresh Picker work, `parseDuration` should return a `Duration` object, which which should have _number_ values. However, because of a Typescript quirk, it returns its values as strings instead.

This means that the following code does not work:

```typescript
import { Duration, formatDuration } from 'date-fns';

const duration: Duration = parseDuration("1d");
const label = formatDuration(duration);
```

`parseDuration` should be fixed to return the type that it says it does (an object with numbers as keys), but I'm not sure what the impacts of that are (but I'm also not sure what hidden bugs we have in our current code due to incorrect types)

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Minor refactor of parseDuration so it returns the correct type.

As a side effect, this fixes a minor bug in the Refresh Picker where the label would only include one unit of duration, if the refresh interval is specified as something like `1m 30s`, because only the first value was converted to a number to format, and the remaining strings were ignored.

**Which issue(s) this PR fixes**:

Fixes #41558

**Special notes for your reviewer**:

I did a smoke test and things seemed fine - the dashboard refresh picker works as expected.

### CODE REVIEW

File: packages/grafana-data/src/datetime/durationutil.test.ts [hunk 1]
```diff
@@ -18,12 +18,12 @@ describe('Duration util', () => {
   describe('parseDuration', () => {
     it('parses a duration string', () => {
       const durationString = '3M 5d 20m';
-      expect(parseDuration(durationString)).toEqual({ months: '3', days: '5', minutes: '20' });
+      expect(parseDuration(durationString)).toEqual({ months: 3, days: 5, minutes: 20 });
```
Comment (mckn) [inline packages/grafana-data/src/datetime/durationutil.test.ts:21]: Could this cause issues for anyone using this util and expecting it to be a string?

Comment (joshhunt) [reply packages/grafana-data/src/datetime/durationutil.test.ts:21]: Yeah, this is a tricky one.

Currently, the function has a bug Its type indicates it returns numbers, but it actually returns strings. So no one _should_ be expecting strings. However, it does return strings. So 🤷🏻 

I'm happy to punt this to 9.0 if we think that's safer?

Comment (mckn) [reply packages/grafana-data/src/datetime/durationutil.test.ts:21]: Cool, lets merge this since it is a bug fix.

**Review by ashharrison90 — APPROVED** [review]
\*chef's kiss\*

**Pull request merged**